### PR TITLE
3452/refactor/realtime account validation

### PR DESCRIPTION
**Pull request by mekarpeles**
Refactors #3452

### Stakeholders

@tabshaikh

### CODE REVIEW

File: openlibrary/plugins/upstream/account.py [hunk 1]
```diff
@@ -402,6 +403,43 @@ def GET(self):
...
 
+class account_validation(delegate.page):
+    path = '/account/validate'
+
+    @staticmethod
+    def validate_username(username):
+        if not username.isalnum():
+            return 'Username may only contain numbers and letters'
+        if not (21 > len(username) > 2):
+            return 'Username must be between 3-20 characters'
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/account.py:414]: We probably want to use the same regex as in other username checks (I believe we used to allow `-` and `_`)

Comment (mekarpeles) [reply openlibrary/plugins/upstream/account.py:414]: Confirmed: https://github.com/internetarchive/openlibrary/pull/3472/files#diff-64069c0678ce3631f1e8346c6630cceaR31

File: openlibrary/plugins/upstream/forms.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def find_ia_account(email=None):
 email_domain_not_blocked = Validator(_("Your email provider is not recognized."), lambda email: not spamcheck.is_spam_email(email))
 username_validator = Validator(_("Username already used"), lambda username: not find_account(lusername=username.lower()))
 
-vlogin = RegexpValidator(r"^[A-Za-z0-9-_]{3,20}$", _('Must be between 3 and 20 letters and numbers'))
+vlogin = RegexpValidator(r"^[A-Za-z0-9-_]{3,20}$", _('Letters and numbers only please, and at least 3 characters.'))
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/forms.py:31]: This regex should be extracted into a variable and used on the backend `/account/validate`

File: openlibrary/plugins/upstream/account.py (hunk 1 above)
Comment (tabshaikh) [reply openlibrary/plugins/upstream/account.py:414]: Okay then will update it

**Comment by tabshaikh** [comment]
@mekarpeles Wouldn't we want to seperate the public reading log and real time signup validation into different pr's

File: openlibrary/plugins/upstream/account.py [hunk 3]
```diff
@@ -402,6 +403,43 @@ def GET(self):
...
         return render['account/verify/failed']()
 
+class account_validation(delegate.page):
+    path = '/account/validate'
+
+    @staticmethod
+    def validate_username(username):
+        if not username.isalnum():
+            return 'Username may only contain numbers and letters'
+        if not (21 > len(username) > 2):
```
Comment (cclauss) [inline openlibrary/plugins/upstream/account.py:413]:
```suggestion
        if not 3 <= len(username) <= 20:
```
Allow the code to mirror the text on the next line.

Comment (tabshaikh) [reply openlibrary/plugins/upstream/account.py:413]: Will do it thanks :)

**Pull request closed without merging**